python-test-samples/apigw-lambda-dynamodb/tests/unit/mock_test.py

```diff
@@ -0,0 +1,135 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0
+
+from os import environ
+import json
+from datetime import datetime
+from unittest import TestCase
+from typing import Any, Dict
+import yaml
+import boto3
+from boto3.dynamodb.conditions import Key
+import moto
+
+# Import the handler under test
+from src import app
+
+# Mock the DynamoDB Service during the test
+@moto.mock_dynamodb
+
+class TestSampleLambdaWithDynamoDB(TestCase):
+    """
+    Unit Test class for src/app.py
+    """
+    
+    def read_sam_template(self, sam_template_fn : str = "template.yaml" ) -> dict:
+        """
+        Utility Function to read the SAM template for the current project
+        """
+        with open(sam_template_fn, "r") as fp:
+            template =fp.read().replace("!","")   # Ignoring tags
+            try:
+                return yaml.safe_load(template)
+            except yaml.YAMLError as exc:
+                    print(exc)
```

Comment:
Did you intend to swallow the exception here?  I guess line 58 will error out with a `KeyError` but it might be more friendly to throw an exception with the actual error.

Reply:
Thank you - probably best just to drop the try-except here.